consensus/blake3/consensus.go

```diff
@@ -861,10 +857,10 @@ func verifyInsideLocation(location []byte, number []*big.Int, config *params.Cha
 
 // Verifies that Location is valid inside current MapContext ontology.
 func checkInsideCurrent(regionLoc int, zoneLoc int, ontology []int) error {
-	if len(ontology) < regionLoc {
+	if regionLoc < 0 || regionLoc > ontology[0] {
 		return consensus.ErrInvalidOntology
 	}
-	if ontology[regionLoc-1] < zoneLoc {
+	if zoneLoc < 0 || zoneLoc > ontology[1] {
```

Comment:
Do you mean `<= 0`? Fuller is given as:
```
FullerOntology   = []int{3, 3}
```
IIUC, this means `0` must be invalid or Fuller defines a 4x4 ontology

Reply:
Agreed.  Will change.

Reply:
We actually mean to compare to 1 as location is also one indexed.